lib/components/room-upgrade-handler.js

```diff
@@ -0,0 +1,135 @@
+const log = require("./logging").get("RoomUpgradeHandler");
+const MatrixRoom = require("../models/rooms/matrix");
+
+/**
+ * Handles migration of rooms when a room upgrade is performed.
+ */
+class RoomUpgradeHandler {
+    /**
+     * @param {RoomUpgradeHandler~Options} opts
```

Comment:
what are they?

Reply:
See below

Reply:
They should be described here in the jsdoc if we expect people to create this object.

Reply:
that's fair, I thought this was much higher in the stack. Do we even need jsdoc here?

Reply:
Yes, otherwise we don't know the type that we are passing.